dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -52,6 +55,8 @@
 
 public class SiteSearchJobImpl {
```

Comment:
Is a synchronization block enough? At the end of the day, the SiteSearchJob in quartz should be a `Stateful` job.  A `Stateful` job can only be run one at a time in _a whole cluster._ This is an important distinction as for most of our Quartz Jobs we do not want them to be run simultaneously in a clustered environment by different servers.

Reply:
Let's say you schedule 2 jobs to run every  30mins (Even if they're meant to work on a different index). They will eventually sync-up and start to use the same code at the same time leading to tons of issues.  for such case synchronization is required.